### ISSUE TITLE
Upgrade eslint to V9 and typescript-eslint v7

### ISSUE THREAD

**Issue by tianyingchun** [belgattitude/nextjs-monorepo-example#5086]
upgrade eslint typescript to https://typescript-eslint.io/blog/announcing-typescript-eslint-v7

**Comment by belgattitude** [comment]
Thanks @tianyingchun It's actually updated, see https://github.com/belgattitude/nextjs-monorepo-example/blob/main/packages/eslint-config-bases/package.json

But be aware than it might cause issues, I'm tracking it down in an another repo https://github.com/belgattitude/shared-dx/pull/539

**Comment by tianyingchun** [comment]
we don't use `typescript-eslint` package directly?

**Comment by belgattitude** [comment]
I think this will answer your question

-  issue https://github.com/vercel/next.js/pull/62137 
- pr - https://github.com/vercel/next.js/pull/62137

**Comment by tianyingchun** [comment]
`typescript-eslint` has `dependencies` 
  "dependencies": {
    "@typescript-eslint/eslint-plugin": "7.0.1",
    "@typescript-eslint/parser": "7.0.1"
  },

 https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-eslint/package.json#L56

**Comment by belgattitude** [comment]
Yes... you're well talking about this repo ?

it's on v7: https://github.com/belgattitude/nextjs-monorepo-example/blob/091ae3bbc29ea8e3141a8bea7cc45cf711b59398/packages/eslint-config-bases/package.json#L70

The warning I mentionned is that it will cause issues till other plugins allows it at v7 too. See above.

Would suggest to play with `yarn why -R @typescript-eslint/parser` to understand the side-effects. A major must be allowed by all other plugins, otherwise the hoisting will lead to eslint 'cannot ... uniquely'

**Comment by tianyingchun** [comment]
yes, agree :) maybe we need to wait the `eslint related plugins upgrade to support V7`

**Comment by tianyingchun** [comment]
eslint V9 now released, will `eslint-config-bases` updated ?

**Comment by belgattitude** [comment]
I’ll wait till nextjs supports it. Feel free to open a pr too. Might help to discover issues as well.

**Comment by belgattitude** [comment]
Btw the pr  https://github.com/vercel/next.js/pull/62137 landed in nextjs 14.2.0

**Comment by belgattitude** [comment]
To keep reference:

- [x] Latest eslint-plugin-react-hooks partially support eslint v9 (no falt config) in https://github.com/facebook/react/pull/28773. Released in v4.6.2. Changelog not clear enough to see if it can replace the 5-canary required by nextjs plugin (TBD)
- [ ] Nextjs eslint plugin, does not yet support v9. (keep track in nextjs issues)

**Comment by tianyingchun** [comment]
waiting few month later maybe :)

**Comment by tianyingchun** [comment]
https://github.com/ArnaudBarre/eslint-plugin-react-refresh
if we need to include `eslint-plugin-react-refresh`

**Comment by belgattitude** [comment]
FWIW updated typescript eslint v8 in https://github.com/belgattitude/nextjs-monorepo-example/pull/6025
